edb/edgeql/compiler/stmt.py

```diff
@@ -286,10 +269,73 @@ def compile_insert_unless_conflict(
     module_id = schema.get_global(
         s_mod.Module, ptr.get_name(schema).module).id
 
+    # Find the source expression corresponding to our field
```

Comment:
Hm. My theory was that doing it before compiling the shape would be a lot easier than trying to pick it out afterward. I'll take a look at building the hoisting in, though that is also a little unsatisfying.

Reply:
Using a default with UNLESS CONFLICT ELSE is a bit dodgy, though, isn't it?

I guess it's semantically totally sensible even if unlikely to be useful.

Reply:
Yes, it's unlikely to be useful, but we have to take it into account nonetheless.

Reply:
I think what I am going to do for this PR is to move the validation after shape compilation and just drop the preliminary hoisting attempt, since it doesn't work in this PR anyway and needs to be rethought

Reply:
Sounds good to me.